File: 2016/day01/day01_test.go
package day01

import (
	"testing"
)

func TestSolve(t *testing.T) {
	d := Day{}

	tests := []struct {
		input  string
		expect int
	}{
		{"R2, L3", 5},
		{"R2, R2, R2", 2},
		{"R5, L5, R5, R3", 12},
	}

	for _, test := range tests {
		t.Run(test.input, func(t *testing.T) {
			value, _ := d.Solve(test.input)
			if value != test.expect {
				t.Errorf("Expected %d, got %d", test.expect, value)
			}
		})
	}
}

File: 2021/test/Day01.test.ts
import { describe, expect, test } from "vitest";
import { readLines } from "../src/Common.bs.js";
import { solve } from "../src/Day01.bs.js";

describe("Day 1", () => {
	test("example", () => {
		const input = readLines("test/inputs/day01-example.txt");
		expect(solve(input)).toMatchInlineSnapshot(`
			{
			  "part1": 7,
			  "part2": 5,
			}
		`);
	});

	test("full", () => {
		const input = readLines("test/inputs/day01.txt");
		expect(solve(input)).toMatchInlineSnapshot(`
			{
			  "part1": 1553,
			  "part2": 1597,
			}
		`);
	});
});
